rust/chains/hyperlane-ethereum/src/interchain_security_module.rs

```diff
@@ -106,6 +108,24 @@ where
             Ok(ModuleType::Unused)
         }
     }
+
+    #[instrument]
+    async fn dry_run_verify(
+        &self,
+        message: &HyperlaneMessage,
+        metadata: &[u8],
+    ) -> ChainResult<Option<U256>> {
+        let tx = self.contract.verify(
+            metadata.to_owned().into(),
+            RawHyperlaneMessage::from(message).to_vec().into(),
+        );
+        let (verifies, gas_estimate) = try_join(tx.call(), tx.estimate_gas()).await?;
+        if verifies {
+            Ok(Some(gas_estimate))
+        } else {
+            Ok(None)
+        }
+    }
```

Comment:
does the normal `call` not return gas used?

Reply:
I also hoped it would but it doesn't look like it: https://docs.rs/ethers/latest/ethers/middleware/contract/struct.ContractDeploymentTx.html#method.call